include/xgboost/generic_parameters.h

```diff
@@ -21,6 +21,8 @@ struct GenericParameter : public XGBoostParameter<GenericParameter> {
   int nthread;
   // primary device, -1 means no gpu.
   int gpu_id;
+  // gpu page size in external memory mode, 0 means using the default.
+  size_t gpu_page_size;
```

Comment:
Parameter is configurable by users, please don't define it twice.  Make one of them a normal variable.  If we don't want to configure it by user, don't use parameter at all.  As pickling might loss some of these information, dask uses pickle to move booster around workers.

Reply:
It's only defined as a configurable parameter once here, the other one is really just plumbing. For now this is mostly used for testing, but perhaps user may want to set it depending on the GPU memory they have.

Reply:
Em.. we want to do parameter validation, like detecting unused parameters.  This may add some extra difficulties.  Do you think it's possible to set it as a DMatrix parameter instead of a global one?  Maybe another PR? Sorry for nitpicking here.

Reply:
Em.. we want to do parameter validation, like detecting unused parameters.  This may add some extra difficulties.  Do you think it's possible to set it as a DMatrix parameter instead of a global one?  Maybe another PR? Sorry for nitpicking here.

Reply:
Agree, we need to be careful adding global parameters due to upcoming work on serialisation. Unless you see a strong motivation for users tuning this, let's leave it out for now.

Reply:
I'm not quite sure if this is useful for end users. Is there way to make a parameter hidden/internal? It's really useful for the tests since we don't have to build a dataset bigger than 32MB.